app/src/main/java/edu/rpi/shuttletracker/AnnouncementsActivity.kt

```diff
@@ -0,0 +1,58 @@
+package edu.rpi.shuttletracker
+import android.os.Bundle
+import android.view.MenuItem
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.Toolbar
+import kotlinx.android.synthetic.main.activity_announcements.*
+import org.json.JSONArray
+import java.net.URL
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+
+
+class AnnouncementsActivity : AppCompatActivity(){
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_announcements)
+        val thread = Thread(kotlinx.coroutines.Runnable {
+            kotlin.run {
+                val announ = URL("https://shuttletracker.app/announcements")
+                var announString = announ.readText()
+                val announArray = JSONArray(announString)
+                val currentDate: LocalDateTime = LocalDateTime.now(ZoneOffset.UTC)
+                val format = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss'Z'")
+
+                announString = ""
+                for (i in 0 until announArray.length()) {
+                    val sched_type = announArray.getJSONObject(i).getString("scheduleType")
+                    val start = announArray.getJSONObject(i).getString("start")
+                    val end = announArray.getJSONObject(i).getString("end")
+                    val startDate = LocalDateTime.parse(start, format)
+                    val endDate = LocalDateTime.parse(end, format)
+                    if((currentDate>=startDate && currentDate<=endDate) || (sched_type =="none") ||
+                        (sched_type =="startOnly" && currentDate>=startDate) ||
+                        (sched_type =="endOnly" && currentDate<=endDate))
+                        announString += startDate.toString().substringBefore('T') + ":\n" +
+                                "______________________________________________"+ "\n" +
+                                announArray.getJSONObject(i).getString("body") + "\n" +
+                                "______________________________________________"+ "\n\n\n"
```

Comment:
I’m amused that these separators are literally just strings of hyphens! It might be a good idea to explore in the future using a proper divider-like UI element for this.